mountpoint-s3/src/fs.rs

```diff
@@ -308,7 +308,13 @@ where
                 return Err(libc::EINVAL);
             }
 
-            let handle = self.superblock.write(&self.client, ino, lookup.inode.parent()).await?;
+            let handle = match self.superblock.write(&self.client, ino, lookup.inode.parent()).await {
+                Ok(handle) => handle,
+                Err(e) => {
+                    error!("open failed: {e:?}");
```

Comment:
(subjective) nit: this is a lot of ceremony just for a log. I'd be tempted to use `map_err` or look for/add something like `on_err`.